src/api/spec/models/bs_request_spec.rb

```diff
@@ -456,4 +456,27 @@
       end
     end
   end
+
+  describe '#obsolete?' do
+    BsRequest::OBSOLETE_STATES.each do |state|
+      context "called for a request with state #{state}" do
+        before do
+          submit_request.update(state: state)
+          submit_request.update(superseded_by: 42) if state == :superseded
+        end
+
+        it { expect(submit_request.reload.obsolete?).to be true }
+      end
+    end
+
+    (BsRequest::VALID_REQUEST_STATES - BsRequest::OBSOLETE_STATES).each do |state|
```

Comment:
I didn't want to squash the two tests, because that would have made it harder to read / understand.

If you mean the performance... the test took 4.0s on my machine. Which seems ok to me.

Anyhow, I checked the code again and realized that this could be handled by a scope and updated the PR.